src/common/utils/shared.rs

```diff
@@ -28,3 +32,78 @@ pub fn adjust_to_size(s: &str, rows: usize, columns: usize) -> String {
         .collect::<Vec<_>>()
         .join("\n\r")
 }
+
+// Colors
+pub mod colors {
+    pub const WHITE: (u8, u8, u8) = (238, 238, 238);
+    pub const GREEN: (u8, u8, u8) = (175, 255, 0);
+    pub const GRAY: (u8, u8, u8) = (68, 68, 68);
+    pub const BRIGHT_GRAY: (u8, u8, u8) = (138, 138, 138);
+    pub const RED: (u8, u8, u8) = (135, 0, 0);
+    pub const ORANGE: (u8, u8, u8) = (215,95,0);
+    pub const BLACK: (u8, u8, u8) = (0, 0, 0);
+}
+
+pub fn hex_to_rgb(hex: &Option<String>) -> (u8, u8, u8) {
+    let c = hex.clone();
+    let imm_str = &c.unwrap();
+    let hex_str: &str = &imm_str;
+    let rgb = Rgb::from_hex_str(hex_str).unwrap().as_tuple();
+    (rgb.0 as u8, rgb.1 as u8, rgb.2 as u8)
+}
+
+// Dark magic
+pub fn detect_theme(bg: (u8, u8, u8)) -> Theme {
+    let (r, g, b) = bg;
+    // HSP, P stands for perceived brightness
+    let hsp: f64 = (0.299 * (r as f64 * r as f64)
+        + 0.587 * (g as f64 * g as f64)
+        + 0.114 * (b as f64 * b as f64))
+        .sqrt();
+    match hsp > 127.5 {
+        true => Theme::Light,
+        false => Theme::Dark,
+    }
+}
+
+pub fn load_palette() -> Palette {
+    let palette = match Colors::new("xresources") {
```

Comment:
@denismaxim0v - my guess is that if you move this call to `OsApi` and then implement something that returns the defaults for the test fixtures, then the tests should pass.

Let me know if you want more detailed instructions :)

Reply:
Well, that makes sense, I will have a go at it this week hopefully